packages/api-explorer/src/Doc.jsx

```diff
@@ -25,44 +26,178 @@ const { Operation } = Oas;
+    this.onModeChange = this.onModeChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
+
+    this.openTutorial = this.openTutorial.bind(this);
     this.operation = this.getOperation();
     this.Params = createParams(this.oas, this.operation);
+
+    this.resetForm = this.resetForm.bind(this);
+
     this.toggleAuth = this.toggleAuth.bind(this);
     this.waypointEntered = this.waypointEntered.bind(this);
-    this.openTutorial = this.openTutorial.bind(this);
-    this.closeTutorialModal = this.closeTutorialModal.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.shouldEnableRequestBodyJsonEditor()) {
+      return;
+    }
+
+    this.operation
+      .getRequestBodyExamples()
+      .then(examples => {
+        if (!Object.keys(examples).length) {
+          return;
+        }
+
+        const jsonExamples = examples.filter(
+          ex => matchesMimeType.json(ex.mediaType) || matchesMimeType.wildcard(ex.mediaType)
+        );
+
+        if (jsonExamples.length) {
+          const example = jsonExamples[0];
+          let code = false;
+
+          if (example.code) {
+            code = example.code;
+          } else if (example.multipleExamples) {
+            code = example.multipleExamples[0].code;
+          }
+
+          try {
+            // Examples are stringified when we get them from `oas` because they need to be stringified for
+            // `@readme/syntax-highlighter` but because we need to pass a usable non-stringified object/array/primitive
+            // to our `CodeSample` component and `@readme/oas-to-har` we're parsing it out here. Cool? Cool.
+            code = JSON.parse(code);
+          } catch (e) {
+            code = {};
+          }
+
+          code = code || {};
+
+          this.setState({
+            formDataJson: code,
+            formDataJsonOriginal: code,
+            formDataJsonRaw: stringifyPretty(code),
+          });
+        }
+      })
+      .catch(() => {
+        // If we fail to generate examples for whatever reason fail silently.
+      });
+  }
+
+  resetForm() {
+    this.setState(previousState => {
+      return {
+        dirty: false,
+        formDataJson: previousState.formDataJsonOriginal,
+        formDataJsonRaw: stringifyPretty(previousState.formDataJsonOriginal),
+        validationErrors: {
+          ...previousState.validationErrors,
+          json: false,
+        },
+      };
+    });
   }
 
   onChange(formData) {
     this.setState(previousState => {
       return {
-        formData: { ...previousState.formData, ...formData },
         dirty: true,
+        formData: { ...previousState.formData, ...formData },
+        validationErrors: {
+          ...previousState.validationErrors,
+          form: false,
+        },
       };
     });
   }
 
+  onJsonChange(rawData) {
+    this.setState(previousState => {
+      let data;
+      try {
+        data = JSON.parse(rawData);
+
+        return {
+          dirty: true,
```

Comment:
actually we might need to still split up the dirty flag and let the form and json editor maintain their own dirty states because right now if you edit the json tab and switch over to the form it'll be dirty despite nothing being changed in it.